src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -818,21 +820,34 @@ func (c controller) isOperatorAvailableInCluster(operatorName string) bool {
 		return false
 	}
 
-	operatorStatus, operatorMessage := utils.ClusterOperatorConditionsToMonitoredOperatorStatus(co.Status.Conditions)
-	err = c.ic.UpdateClusterOperator(context.TODO(), c.ClusterID, operatorName, operatorStatus, operatorMessage)
+	operatorStatusInService, err := c.ic.GetClusterMonitoredOperator(utils.GenerateRequestContext(), c.ClusterID, operatorName)
 	if err != nil {
-		c.log.WithError(err).Warnf("Failed to update %s operator status %s with message %s", operatorName, operatorStatus, operatorMessage)
+		c.log.WithError(err).Errorf("Failed to get cluster %s operator %s status", c.ClusterID, operatorName)
 		return false
 	}
 
-	if !c.checkOperatorStatusCondition(co, configv1.OperatorAvailable, configv1.ConditionTrue) ||
-		!c.checkOperatorStatusCondition(co, configv1.OperatorDegraded, configv1.ConditionFalse) {
-		return false
-	}
+	operatorStatus, operatorMessage := utils.ClusterOperatorConditionsToMonitoredOperatorStatus(co.Status.Conditions)
+	if operatorStatusInService.Status != operatorStatus || (operatorStatusInService.StatusInfo != operatorMessage && operatorMessage != "") {
+		c.log.Infof("Operator %s status: %s message: %s", operatorName, operatorStatus, operatorMessage)
 
-	c.log.Infof("%s operator is available in cluster", operatorName)
+		err = c.ic.UpdateClusterOperator(context.TODO(), c.ClusterID, operatorName, operatorStatus, operatorMessage)
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to update %s operator status %s with message %s", operatorName, operatorStatus, operatorMessage)
+			return false
+		}
```

Comment:
Consider extracting to a separate method and call it from waitForOLMOperators and here

Reply:
I opened a ticket for merging those
https://issues.redhat.com/browse/MGMT-7149